From 649d03adaebfc929243ad73b5f4178818c9b0461 Mon Sep 17 00:00:00 2001
From: Georges Basile Stavracas Neto <georges.stavracas@gmail.com>
Date: Wed, 13 Apr 2016 22:57:54 -0300
Subject: year-view: check for desktop environment

This is a work around the crash introduced when we
started depending on a setting that is not our own.

Since we can't do that anymore for 3.20, let's just
check for the desktop environment and avoid the crash.
---
 src/gcal-year-view.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

Index: gnome-calendar-3.20.2/src/gcal-year-view.c
===================================================================
--- gnome-calendar-3.20.2.orig/src/gcal-year-view.c
+++ gnome-calendar-3.20.2/src/gcal-year-view.c
@@ -1494,10 +1494,14 @@ gcal_year_view_init (GcalYearView *self)
   self->end_selected_date = g_new0 (icaltimetype, 1);
   self->end_selected_date->zone = e_cal_util_get_system_timezone ();
 
-  /* bind GNOME Shell' show week numbers property to GNOME Calendar's one */
-  self->shell_settings = g_settings_new ("org.gnome.shell.calendar");
-  g_settings_bind (self->shell_settings, "show-weekdate", self, "show-week-numbers", G_SETTINGS_BIND_DEFAULT);
-  g_signal_connect_swapped (self->shell_settings, "changed::show-weekdate", G_CALLBACK (gtk_widget_queue_draw), self);
+  /* FIXME: this setting should be moved to gsettings-desktop-schemas */
+  if (g_strcmp0 (g_getenv ("XDG_CURRENT_DESKTOP"), "GNOME") == 0)
+    {
+      /* bind GNOME Shell' show week numbers property to GNOME Calendar's one */
+      self->shell_settings = g_settings_new ("org.gnome.shell.calendar");
+      g_settings_bind (self->shell_settings, "show-weekdate", self, "show-week-numbers", G_SETTINGS_BIND_DEFAULT);
+      g_signal_connect_swapped (self->shell_settings, "changed::show-weekdate", G_CALLBACK (gtk_widget_queue_draw), self);
+    }
 
   gtk_list_box_set_header_func (GTK_LIST_BOX (self->events_sidebar), update_sidebar_headers, self, NULL);
   gtk_list_box_set_sort_func (GTK_LIST_BOX (self->events_sidebar), sidebar_sort_func, NULL, NULL);
